megatron/initialize.py

```diff
@@ -274,3 +274,15 @@ def _initialize_mem_buffs():
     # Initialize memory for checkpointed activations.
     if args.distribute_checkpointed_activations:
         mpu.init_checkpointed_activations_memory_buffer()
+
+def _is_rank_0():
+    """Check whether it is rank 0. For AML, check if it is rank 0 of a node"""
+    if torch.distributed.is_initialized():
+        if torch.distributed.get_rank() == 0 or (
+            'AZUREML_EXPERIMENT_ID' in os.environ and torch.distributed.get_rank() % torch.cuda.device_count() == 0
```

Comment:
can we use is_aml() function here to simplify the if check

Reply:
__init__.py requires this initialize.py so it's having circular dependencies if I try to use it. Is there another place I can put those?